main.go

```diff
@@ -3,94 +3,232 @@ package main
 import (
 	"context"
 	"crypto/tls"
+	"encoding/json"
 	"flag"
 	"fmt"
 	"net/http"
 	"os"
+	"os/signal"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
 
 	"github.com/grafana/detect-angular-dashboards/api"
 	"github.com/grafana/detect-angular-dashboards/api/gcom"
 	"github.com/grafana/detect-angular-dashboards/api/grafana"
 	"github.com/grafana/detect-angular-dashboards/build"
 	"github.com/grafana/detect-angular-dashboards/detector"
+	"github.com/grafana/detect-angular-dashboards/flags"
 	"github.com/grafana/detect-angular-dashboards/logger"
 	"github.com/grafana/detect-angular-dashboards/output"
 )
 
-const (
-	envGrafanaToken = "GRAFANA_TOKEN"
-)
+const envGrafana = "GRAFANA_TOKEN"
 
-func newLogger(verboseFlag, jsonOutputFlag bool) *logger.LeveledLogger {
-	log := logger.NewLeveledLogger(verboseFlag)
-	if jsonOutputFlag {
-		// Redirect everything to stderr to avoid mixing with json output
-		log.Logger.SetOutput(os.Stderr)
-		log.WarnLogger.SetOutput(os.Stderr)
-	}
-	return log
-}
-
-func getToken() (string, error) {
-	token := os.Getenv(envGrafanaToken)
-	if token == "" {
-		return "", fmt.Errorf("missing env var %q", envGrafanaToken)
-	}
-	return token, nil
+type Output struct {
+	mu   sync.Mutex
+	data []output.Dashboard
 }
 
 func main() {
-	versionFlag := flag.Bool("version", false, "print version number")
-	verboseFlag := flag.Bool("v", false, "verbose output")
-	jsonOutputFlag := flag.Bool("j", false, "json output")
-	skipTLSFlag := flag.Bool("insecure", false, "skip TLS verification")
-	flag.Parse()
+	flags := flags.ParseFlags()
 
-	if *versionFlag {
+	if flags.Version {
 		fmt.Printf("%s %s (%s)\n", os.Args[0], build.LinkerVersion, build.LinkerCommitSHA)
 		os.Exit(0)
 	}
-	log := newLogger(*verboseFlag, *jsonOutputFlag)
+	log := newLogger(flags.Verbose, flags.JSONOutput)
 
 	token, err := getToken()
 	if err != nil {
-		_, _ = fmt.Fprintln(os.Stderr, err.Error())
+		log.Errorf("Failed to retrieve Grafana token: %s\n", err.Error())
 		os.Exit(1)
 	}
+	client := initializeClient(token, &flags)
+
+	ticker := time.NewTicker(flags.Interval)
+	defer ticker.Stop()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Handle graceful shutdown
+	setupGracefulShutdown(cancel, ticker, log)
+
+	// Channel used to send runDetection results to the HTTP server
+	outputChan := make(chan []output.Dashboard)
+	output := &Output{}
+
+	// Readiness flag using atomic boolean
+	var ready int32
+	var once sync.Once
+
+	go func() {
+		for data := range outputChan {
+			log.Log("Updating outputData with results from most recent detection")
+
+			// Need to lock to avoid concurrent read/write access to outputData
+			output.mu.Lock()
+			output.data = data
+			output.mu.Unlock()
+
+			// Use sync.Once to set readiness only once
+			once.Do(func() {
+				atomic.StoreInt32(&ready, 1)
+				log.Log("Updating readiness probe to ready")
+			})
+		}
+	}()
+
+	if flags.ServerMode {
```

Comment:
Thank you for this! This was on my list to address, but I got distracted with other work. The PR looks good - but since we've gone back to using the `detector.Run` function, I fear this may re-introduce the JSONOutputter.Output duplicate bug. Happy to merge your PR into this branch once we have that addressed.

Reply:
Sorry I didn't  read your PR correctly. It worked as expected and I was able to merge it.